src/common/snippets/src/lowered/pass/insert_specific_iterations.cpp

```diff
@@ -75,12 +75,29 @@ LoopManager::LoopBounds InsertSpecificIterations::insert_copy_loop(LinearIR& lin
                                                                    std::vector<LoopPort>& new_entry_ports, std::vector<LoopPort>& new_exit_ports) {
     const auto& loop_manager = linear_ir.get_loop_manager();
     const auto loop_bounds = loop_manager->get_loop_bounds(linear_ir, loop_id);
+    const auto loop_begin_pos = loop_bounds.first;
+    const auto loop_end_pos = std::next(loop_bounds.second);
+
     ExpressionMap expression_map;
     const auto& cloning_config = LinearIRBuilder::Config(false);
-    const auto& loop_copy_range = LinearIRBuilder(cloning_config).clone_range(loop_bounds.first, std::next(loop_bounds.second), expression_map);
+    const auto& loop_copy_range = LinearIRBuilder(cloning_config).clone_range(loop_begin_pos, loop_end_pos, expression_map);
     const auto new_loop_begin_pos = linear_ir.insert(insert_pos, loop_copy_range.begin(), loop_copy_range.end());
     const auto new_loop_end_pos = std::prev(insert_pos);
 
+    // Added connections between output of cloned bodies and the current LinearIR
+    for (LinearIR::constExprIt result_it = new_loop_begin_pos, original_it = loop_begin_pos; original_it != loop_end_pos; ++result_it, ++original_it) {
+        const auto result_expr = *result_it;
+        const auto original_expr = *original_it;
+        for (size_t i = 0; i < original_expr->get_output_count(); i++) {
+            const auto& consumers = original_expr->get_output_port_connector(i)->get_consumers();
+            for (const auto& consumer : consumers) {
+                if (std::find(loop_begin_pos, loop_end_pos, consumer.get_expr()) == loop_end_pos) {
+                    result_expr->get_output_port_connector(i)->add_consumer(consumer);
+                }
+            }
+        }
+    }
+
```

Comment:
So how it used to worked before?
Do I understand correctly tha specific iterations were "stacked" in the LIR, but not connected previously?
And why do we need these connections for dynamic buffers?

Reply:
Added buffers with multiple inputs for that in https://github.com/openvinotoolkit/openvino/pull/24646/commits/ed75da847b7705e404e8b48f611c352c55590a2e